arrow-optics/src/main/kotlin/arrow/optics/extensions/either/each/EitherEach.kt

```diff
@@ -0,0 +1,45 @@
+package arrow.optics.extensions.either.each
+
+import arrow.core.Either
+import arrow.core.Either.Companion
+import arrow.optics.PTraversal
+import arrow.optics.extensions.eitherEach
+import arrow.optics.typeclasses.Each
+import kotlin.Any
+import kotlin.Deprecated
+import kotlin.PublishedApi
+import kotlin.Suppress
+import kotlin.jvm.JvmName
+
+/**
+ * cached extension
+ */
+@PublishedApi()
+internal val each_singleton: Each<Either<Any?, Any?>, Any?> = eitherEach()
+
+@JvmName("each")
+@Suppress(
+  "UNCHECKED_CAST",
+  "USELESS_CAST",
+  "EXTENSION_SHADOWED_BY_MEMBER",
+  "UNUSED_PARAMETER"
+)
+@Deprecated(
+  "@extension kinded projected functions are deprecated. Use Either.each().each() instead",
+  ReplaceWith("Either.each<L, R>().each()", "arrow.core.Either", "arrow.optics.each"),
+  DeprecationLevel.WARNING
+)
+fun <L, R> each(): PTraversal<Either<L, R>, Either<L, R>, R, R> = arrow.core.Either
```

Comment:
Initially, I'm removing this method in favor of `Either.each<L, R>().each()` because we'll have a name collision with the same method for `Option` since we're moving them to the `arrow.optics` package.

Another alternative could be adding this method to the Either's companion object but we'll get another name collision with the method that returns the `Each` instance for `Either`

Reply:
I think we can deprecate `Each`, and recommend to use `Traversal` directly. This would get rid of this name collision. We would also have to update [this file](https://github.com/arrow-kt/arrow-optics/blob/master/arrow-optics/src/main/kotlin/arrow/optics/dsl/each.kt) to have the same APIs but for `Traversal` directly and deprecate them as well.

Deprecating `Each` can be done in seperate PR 👍 
Then we just have to update the deprecation notice to:

```
@Deprecated(
  "Each is being deprecated. Use `Traversal` directly instead.",
  ReplaceWith("Either.traversal<L, R>()", "arrow.core.Either", "arrow.optics.traversal"),
  DeprecationLevel.WARNING
)
```

Reply:
Sounds good. I'm gonna deprecate each methods for Either in this pull request and I'll deprecate Each in a separate PR